### ISSUE TITLE
Functions at 0x0 get bad decompilation

### ISSUE THREAD

**Issue by jrmuizel** [NationalSecurityAgency/ghidra#386]
Building the following with clang and -O2 on macos:
```c++
class Foo {
  int f(int a);
};

int Foo::f(int a) { return a; }
```
gives:
[test.o.zip](https://github.com/NationalSecurityAgency/ghidra/files/3056557/test.o.zip)

which incorrectly decompiles to:

```c

/* Foo::f(int) */

ulong __thiscall f(Foo *this,int param_1)

{
  uint in_ESI;
  
  return (ulong)in_ESI;
}
```

It should be more like:
```c

/* Foo::f(int) */

ulong __thiscall f(Foo *this,int param_1)

{
  return (ulong)(uint)param_1;
}
```

**Comment by emteere** [comment]
Looks like you've found an edge case with the function information stored in the program database that is stored at address 0x0, and then passed to the decompiler.  The error only occurs for me if the function signature is committed from the decompiler.  Are you seeing this behavior elsewhere than 0x0?

Also a stack trace in the function start patterns, most likely because of the pattern found at 0x0.

This should be fixed as you might be loading many '.o' files to extract information and the '.o' hasn't been moved to it's final resting place.

Thanks for the bug, we'll take a closer look!

**Comment by jrmuizel** [comment]
Yeah, I confirmed that it's specific to the function being at 0x0

**Comment by emteere** [comment]
Thanks for checking into it further.

**Comment by emteere** [comment]
All issues should be addressed by 9.2 and previous decompiler fixes
